client/src/reducers/metadata.js

```diff
@@ -0,0 +1,22 @@
+const Metadata = (
+  state = {
+    singleValueCategories: new Map(),
+  },
+  action
+) => {
+  const { category, value } = action;
+  const { singleValueCategories } = state;
+  switch (action.type) {
+    case "add singleValueCategory":
+      if (value)
+        return {
+          ...state,
+          singleValueCategories: singleValueCategories.set(category, value),
```

Comment:
1. Is this reducer going to be triggered, and the data structure reset, each time categories updates? 
2. Is this aware of user annotated categories, where `unassigned` is the only value? 

<img width="1088" alt="Screen Shot 2020-09-01 at 11 46 02 AM" src="https://user-images.githubusercontent.com/1770265/91874876-bebbca80-ec48-11ea-9969-3e25df068518.png">

Reply:
1. The Redux store holds a map which adds a key, value pair for any singletons dispatched.  It's never "reset" as `singleValueCategories` will always be pointing to the same object.

2. Yes, the dispatch where these singletons are added is in place of where we would render the modified LSB values.  The check to dispatch is as follows:
```javascript
 const isSingularValue = !isUserAnno && numCategoryValues === 1;
```

Reply:
My hangup here is that it seems like we're dispatching multiple times unnecessarily, even if we're not resetting the value in redux. It doesn't seem like these need to know about each other. 

Why not take in all categorical values in the drawer, and do a check in that component for the categories that only have one value? Then all of this logic is encapsulated in the drawer component rather than being piped through from the LSB.

Reply:
@colinmegill Good thinking, I've just refactored the component into a react-async component. PTAL!